plugin.rb

```diff
@@ -31,33 +31,33 @@ module ::DataExplorer
   def self.plugin_name
     'discourse-data-explorer'.freeze
   end
-
-  def self.pstore_get(key)
-    PluginStore.get(DataExplorer.plugin_name, key)
-  end
-
-  def self.pstore_set(key, value)
-    PluginStore.set(DataExplorer.plugin_name, key, value)
-  end
-
-  def self.pstore_delete(key)
-    PluginStore.remove(DataExplorer.plugin_name, key)
-  end
 end
 
 after_initialize do
+  load File.expand_path('../app/models/data_explorer/query.rb', __FILE__)
+  load File.expand_path('../app/controllers/data_explorer/query_controller.rb', __FILE__)
+  load File.expand_path('../app/serializers/data_explorer/query_serializer.rb', __FILE__)
+  load File.expand_path('../app/serializers/data_explorer/small_badge_serializer.rb', __FILE__)
+  load File.expand_path('../app/serializers/data_explorer/small_post_with_excerpt_serializer.rb', __FILE__)
 
   add_to_class(:guardian, :user_is_a_member_of_group?) do |group|
     return false if !current_user
     return true if current_user.admin?
     return current_user.group_ids.include?(group.id)
   end
 
-  add_to_class(:guardian, :user_can_access_query?) do |group, query|
+  add_to_class(:guardian, :user_can_access_query?) do |query|
+    return false if !current_user
+    return true if current_user.admin?
+    query.groups.blank? || query.groups.any? do |group|
+      user_is_a_member_of_group?(group)
+    end
+  end
+
+  add_to_class(:guardian, :group_and_user_can_access_query?) do |group, query|
     return false if !current_user
     return true if current_user.admin?
-    return user_is_a_member_of_group?(group) &&
-           query.group_ids.include?(group.id.to_s)
+    return user_is_a_member_of_group?(group) && query.groups.where(id: group.id).present?
```

Comment:
```suggestion
    return user_is_a_member_of_group?(group) && query.groups.exists?(id: group.id)
```

`exists?` is slightly more efficient here since we don't have to create an ActiveRecord object.